### PR TITLE
Migrate to ActivityResultContracts

### PR DESCRIPTION
**Pull request by msfjarvis**
## :loudspeaker: Type of change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates

## :scroll: Description
Migrates ~~some~~ nearly all uses of `setTargetFragment` and `startActivityForResult` to `ActivityResultContracts`.

## :bulb: Motivation and Context
`ActivityResultContracts` is a cleaner and more effective abstraction and supports a lot more things than just `startActivityForResult` OOTB.

## :green_heart: How did you test it?
Manually verify app works as before

## :pencil: Checklist

- [x] I formatted the code with the IDE's reformat action (Ctrl + Shift + L/Cmd + Shift + L)
- [x] I reviewed submitted code
- [ ] I added a [CHANGELOG](CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by msfjarvis** [comment]
@FabianHenneke this did resolve our problem with sync not calling `refreshPasswordList` and I also fixed a bug that I missed in #909

**Comment by msfjarvis** [comment]
I'm not going to touch the legacy Autofill implementation, and I can't think of a good way to eliminate the onActivityResult in `BaseGitActivity`.

**Comment by fmeum** [comment]
> I'm not going to touch the legacy Autofill implementation, and I can't think of a good way to eliminate the onActivityResult in `BaseGitActivity`.

Me neither, we can tackle it some time after the GitAsyncTask migration.

**Comment by msfjarvis** [comment]
@FabianHenneke this is good to go, I don't see anything I'd like to change.

File: app/src/main/java/com/zeapo/pwdstore/PasswordStore.kt [hunk 1]
```diff
@@ -756,61 +810,6 @@ class PasswordStore : AppCompatActivity(R.layout.activity_pwdstore) {
...
 
-    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
-        if (resultCode == Activity.RESULT_OK) {
-            when (requestCode) {
-                // if we get here with a RESULT_OK then it's probably OK :)
-                BaseGitActivity.REQUEST_CLONE -> settings.edit { putBoolean(PreferenceKeys.REPOSITORY_INITIALIZED, true) }
-                // if went from decrypt->edit and user saved changes, we need to commitChange
-                REQUEST_CODE_DECRYPT_AND_VERIFY -> {
-                    if (data != null && data.getBooleanExtra("needCommit", false)) {
-                        if (data.getStringExtra("OPERATION") == "EDIT") {
```
Comment (fmeum) [inline app/src/main/java/com/zeapo/pwdstore/PasswordStore.kt:767]: It seems to me that this `onActivityResult` logic has no correspondence anymore, is this intended?

**Review by fmeum — COMMENTED** [review]
I will take a deeper look tomorrow.

File: app/src/main/java/com/zeapo/pwdstore/PasswordStore.kt (hunk 1 above)
Comment (msfjarvis) [reply app/src/main/java/com/zeapo/pwdstore/PasswordStore.kt:767]: We don't have any other occurances of `"needCommit"` in the tree so this never gets hit.

**Review by fmeum — APPROVED** [review]
Went through all the files and found the changes to be consistent. Well done!

**Pull request merged**